app.js

```diff
@@ -0,0 +1,280 @@
+      ? []
+      : JSON.parse(localStorage.getItem(`${todayDateInput.value}done`));
+  newDoneList.forEach((newDone) => {
+    addNewDoneListToShowingUI(newDone);
+  });
+}
+
+// 웹 페이지가 처음 렌더링 되는 시점에 로컬 스토리지로 부터 불러옴. 아직 존재하지 않는다면 빈 배열을 할당(삼항 연산자 이용)
+const firstDomMountedDateFormatString = getNowDateFormatString();
+
+let todoList =
+  localStorage.getItem(`${todayDateInput.value}todo`) === null
+    ? []
+    : JSON.parse(localStorage.getItem(`${todayDateInput.value}todo`));
+let doneList =
+  localStorage.getItem(`${todayDateInput.value}done`) === null
+    ? []
+    : JSON.parse(localStorage.getItem(`${todayDateInput.value}done`));
+
+// todo를 done으로 로컬 스토리지 갱신을 하고 UI에 반영하는 함수
+function moveTodoToDone(itemName) {
+  const todayDate = todayDateInput.value;
+  let prevTodoList = JSON.parse(localStorage.getItem(`${todayDate}todo`));
+  const index = prevTodoList.indexOf(itemName);
+  if (index !== -1) {
+    prevTodoList.splice(index, 1);
+  }
+  localStorage.setItem(`${todayDate}todo`, JSON.stringify(prevTodoList)); // 기존의 로컬스토리지 todo에서 지우는 로직
+
+  let prevDoneList =
+    localStorage.getItem(`${todayDate}done`) === null
+      ? []
+      : JSON.parse(localStorage.getItem(`${todayDate}done`));
+  prevDoneList.push(itemName);
+  localStorage.setItem(`${todayDate}done`, JSON.stringify(prevDoneList)); // done 로컬 스토리지에 반영하는 로직
+
+  getDownAllTodoItemsFromDomAndRemount();
+  getDownAllDoneItemsFromDomAndRemount();
+}
+
+// todo를 로컬 스토리지에서 지워버리고 UI에 반영하는 함수
+function removeFromTodo(itemName) {
+  const todayDate = todayDateInput.value;
+  let prevTodoList = JSON.parse(localStorage.getItem(`${todayDate}todo`));
+  const index = prevTodoList.indexOf(itemName);
+  if (index !== -1) {
+    prevTodoList.splice(index, 1);
+  }
+  localStorage.setItem(`${todayDate}todo`, JSON.stringify(prevTodoList));
+  getDownAllTodoItemsFromDomAndRemount();
+  // done에 대한 연산은 굳이 불필요하므로 리소스 상 하지 않는다
+}
+
+// 새로운 요소를 할일 UI 쪽에 추가하는 함수. 새로운 dom 요소를 생성하여 UI에 반영함
+function addNewTodoListToShowingUI(newTodo) {
+  const showingTodoDiv = document.createElement('div');
+  showingTodoDiv.classList.add('showingTodo');
+  const showingTodoText = document.createElement('p');
+  showingTodoText.classList.add('showingTodoText');
+  showingTodoText.innerText = newTodo.slice(0, newTodo.length - 13);
+  showingTodoDiv.appendChild(showingTodoText);
+
+  const showingCheckImage = document.createElement('img');
+  showingCheckImage.classList.add('checkImage');
+  showingCheckImage.src = 'assets/images/check-solid.svg';
+  showingCheckImage.alt = 'This is check image';
+  showingCheckImage.addEventListener('click', () => {
+    moveTodoToDone(newTodo);
+  });
+
+  const showingTrashCanImage = document.createElement('img');
+  showingTrashCanImage.classList.add('trashCanImage');
+  showingTrashCanImage.src = 'assets/images/trash-solid.svg';
+  showingTrashCanImage.alt = 'This is trash can image';
+  showingTrashCanImage.addEventListener('click', () => {
+    removeFromTodo(newTodo);
+  });
```

Comment:
이 부분에서 상당히 많은 부분이 겹치는 것 같습니다
Image를 만드는 함수를 따로 분리해서 다음과 같이 만들어 보는건 어떨까요?
```suggestion
  function createImage(src, alt, onClick) {
  const image = document.createElement('img');
  image.src = src;
  image.alt = alt;
  image.addEventListener('click', onClick);
  return image;
}
```